File: ecmascript/prime-factors/prime-factors.js
export default class PrimeFactors {
  for(n){
    let primes = [];
    let factor = 2;

    while (factor <= n) {
      if (n % factor === 0) {
        primes.push(factor);
        n = n / factor;
      } else {
        factor++;
      }
    }
    return primes;
  }
}

File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  def self.for(n)
    primes = []
    (2..n).each do |i|
      while n % i == 0 do
        primes.push(i)
        n /= i
      end
      break if i >= n
    end
    primes
  end
end
